### DISCUSSION TITLE
同一进程内，向service发送一个比较大比较复杂得table，有个异常抛出

### DISCUSSION THREAD

**Discussion by zuosixiaoxiongmao** [cloudwu/skynet#1762] [General]
datacenter.lua:433: serialize can't pack too depth table
有时候目标service也不会抛出异常，只会单纯没收到带这table的消息。
请问这种情况有什么好方法解决。

**Comment by hanxi** [comment]
简单粗暴就是把 depth 改大一点，根治还是得从逻辑层减少不必要的数据发送。

**Comment by cloudwu** [comment]
也可能是循环引用了。比如 table 中包含了自己。

**Comment by terry8210** [comment]
递归引用

**Comment by terry8210** [comment]
是不是用了类似class setmetatable包装的结构？
